NMSSH/NMSSHHostConfig.m

```diff
@@ -34,7 +34,7 @@ - (void)mergeFrom:(NMSSHHostConfig *)other {
     if (!self.user) {
         [self setUser:other.user];
     }
-    if (!self.port) {
+    if (![self.port boolValue]) {
```

Comment:
`self.port` is not a boolean, the correct check is `self.port == nil`

Reply:
I thought that I had tried that one first and that it still gave the warning. But I tried it again and now the warning vanished.

Shall I correct the PR or will you directly make the change yourself?

BTW: boolValue seems correct as well:
> The number object's value expressed as a Boolean value.

Reply:
Update the PR and I'll merge it!

Using `boolValue` you would get `false` if `self.port == 0`, so your `if` condition is not equivalent to the previous one.

Reply:
Updated

Reply:
And as another BTW:

> if (!self.port)

would evaluate to FALSE as well if self.port == 0

Reply:
What I meant was that `[self.port boolValue]` returns `false` if `self.port` is `nil` or `0`, 
so the `if` condition would be `true` for `nil` and `0`. While checking for `!self.port` returns `true` only if `self.port` is `nil`.